src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -11,15 +11,34 @@ public class ReadWriteList<E> {
 
     public void add(E element) {
         // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
         // write your code here
-        return null;
+        lock.readLock().lock();
+        try {
+            if (index >= 0 && index < list.size()) {
+                return list.get(index);
+            }
+            return null;
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public int size() {
         // write your code here
-        return 0;
+        lock.readLock().lock();
```

Comment:
You can create some variable `lock.readLock()` to avoid method calling duplicates (the same for other methods.